### DISCUSSION TITLE
Git ssh connecting gets stuck... help!

### DISCUSSION THREAD

**Discussion by tubbadu** [community/community#22053] [New to GitHub]
<p>Hi everyone, I have a strange problem:<br>
if I connect my pc to the ethernet cable and try to do anything with the ssh key (eg. cloning a repository, pushing a commit, …) it gets stuck, without cloning or pushing etc.<br>
instead if I connect to the wifi (of the same exact router!) it works! so strange…<br>
I’ll post here the output of the command <code>ssh -Tv git@github.com</code>:<br>
<a href="https://pastebin.com/NNZqAkt6" rel="noopener nofollow ugc">output when connected with the ethernet cable (gets stuck)</a><br>
<a href="https://pastebin.com/hkXPfcvw" rel="noopener nofollow ugc">output when connected to the wifi (works well)</a></p>
<p>I’m running Kubuntu 22.04, but I have the same problem on a Fedora live usb, so it shouldn’t be a OS problem…</p>
<p>any Idea on what can I do to fix this? http cloning works well, but I need ssh :frowning:<br>
thanks in advance!</p>

**Comment by airtower-luna** [comment]
<p><a class="mention" href="/cronlabspl">@cronlabspl</a>, those error messages are irrelevant, in both cases the host key is found in the user <code>known_hosts</code> file:</p>
<pre><code class="lang-auto">debug1: Host '[ssh.github.com]:443' is known and matches the ED25519 host key.
debug1: Found key in /home/tubbadu/.ssh/known_hosts:5
</code></pre>
<p>What’s odd is that in both cases the connections gets far enough for the server to accept the client key, so it’s not a simple case of a firewall blocking all communication.</p>
<p>Things to try:</p>
<ul>
<li>Increase the verbosity level for SSH (add more <code>-v</code> to the command line). Maybe there’s another relevant detail.</li>
<li>Check the router logs and settings. Is there any kind of difference in how traffic from wifi and ethernet is treated? Does it block ICMP packages (that’d be bad!)? Also check the MTU of relevant devices, if the MTU of the external device is lower than that of the internal ethernet you might need to enable MSS clamping (especially if something’s messing with ICMP).</li>
<li>You seem to be using SSH over the HTTPS port. In principle that should work, but you could see if the regular SSH port works.</li>
</ul>

**Comment by airtower-luna** [comment]
<p>Exactly. Which is why I suggested options for further debugging, instead of sending the person who asked after a red herring.</p>

**Comment by tubbadu** [comment, marked as answer]
<p>finally found a workaround: running</p>
<pre><code class="lang-auto">sudo /sbin/iptables -A OUTPUT -t mangle -p tcp --dport 22 -j TOS --set-tos 0x00
</code></pre>
<p>fixes the problem for the current session</p>
<p>thanks to all!</p>

**Comment by airtower-luna** [comment]
<p>Interesting, thanks for sharing! Looks like QoS settings in the router might be worth a look. :sweat_smile:</p>
